trappy/ftrace.py

```diff
@@ -499,32 +500,35 @@ def __process_path(self, basepath):
             trace_name = os.path.join(basepath, "trace")
 
         trace_txt = trace_name + ".txt"
-        trace_raw = trace_name + ".raw.txt"
         trace_dat = trace_name + ".dat"
 
         if os.path.isfile(trace_dat):
-            # Both TXT and RAW traces must always be generated
-            if not os.path.isfile(trace_txt) or \
-               not os.path.isfile(trace_raw):
+            # TXT traces must always be generated
+            if not os.path.isfile(trace_txt):
                 self.__run_trace_cmd_report(trace_dat)
-            # TXT (and RAW) traces must match the most recent binary trace
+            # TXT traces must match the most recent binary trace
             elif os.path.getmtime(trace_txt) < os.path.getmtime(trace_dat):
                 self.__run_trace_cmd_report(trace_dat)
 
-        if not os.path.isfile(trace_raw):
-            trace_raw = None
+        return trace_txt, None
 
-        return trace_txt, trace_raw
+    def __get_raw_event_list(self):
+        self.raw_events = []
+        # Generate list of events which need to be parsed in raw format
+        for event_class in (self.thermal_classes, self.sched_classes, self.dynamic_classes):
+            for trace_class in event_class.itervalues():
+                raw = getattr(trace_class, 'parse_raw', None)
+                if raw:
+                    name = getattr(trace_class, 'name', None)
+                    if name:
+                        self.raw_events.append(name)
 
     def __run_trace_cmd_report(self, fname):
         """Run "trace-cmd report fname > fname.txt"
```

Comment:
This should probably be updated to be:

```python
"""Run "trace-cmd report [-r raw_event]* fname > fname.txt
```

and we can also add a sentence in the following paragraph explaining the usage of the `-r` flag